### PR TITLE
Templates API Features

### PR DESCRIPTION
**Pull request by Raphkitue**
Introducing our recent new functionnalities

### CODE REVIEW

File: package.json
Comment (dnmeid) [inline package.json:1]: - Need to update version here as well
- please update @companion-module/base
- please update @companion-module/tools

File: src/actions.js [hunk 1]
```diff
@@ -311,25 +312,140 @@ export function getActions() {
...
 		this.log('info', `request ${requestString}`)
-		got
-			.get(requestString)
+		var req
+		if (requestType === 'POST')
+		{
+			this.log('info', 'isPostRequest')
+			const formData = new FormData();
+			req = got.post(requestString, 
+				{ body: formData })
```
Comment (dnmeid) [inline src/actions.js:429]: It seems formData is just created and never modified. So this should always result in the same object. Would it be better to use a const instead?

File: src/feedback.js [hunk 2]
```diff
@@ -321,5 +321,74 @@ export function initFeedbacks() {
...
+			},
+			options: [],
+			callback: ({ options }) => {
+				if (this.data.playoutRunning && this.data.elementsStatuses[this.data.upcomingElementId] === UNAVAILABLE_STATUS) {
+					return true
+				}
+			},
+		}
+		feedbacks.templateInsertStatus = {
+			type: 'advanced',
```
Comment (dnmeid) [inline src/feedback.js]: You can do this with an advanced feedback, but I would suggest to stay with boolean. Users have more flexibility with boolean and they also can be used in triggers.
You can solve this with one feedback resulting to true if `this.data.templateInsertStatus`is same value as option. Then the users have to add the feedback two times with different option value and different colors to visualize al three states. You can offer this from a preset.

**Review by dnmeid — CHANGES_REQUESTED** [review]

File: package.json
Comment (Raphkitue) [reply package.json:1]: Should be good now

File: src/actions.js (hunk 1 above)
Comment (Raphkitue) [reply src/actions.js:429]: Right, changed it to an exported const on top of the file

File: src/feedback.js (hunk 2 above)
Comment (Raphkitue) [reply src/feedback.js]: Thanks for the suggestions, should be correclty applied now

File: src/feedback.js [hunk 3]
```diff
@@ -321,5 +321,67 @@ export function initFeedbacks() {
...
+			callback: ({ options }) => {
+				if (this.data.playoutRunning && this.data.elementsStatuses[this.data.upcomingElementId] === UNAVAILABLE_STATUS) {
+					return true
+				}
+			},
+		}
+		feedbacks.templateInsertStatus = {
+			type: 'boolean',
+			name: 'Status of the recent template insert',
+			description: 'Indicates if a triggered template insertion is running',
```
Comment (dnmeid) [inline src/feedback.js]: I think "...is running" does not describe both of your options is running/has failed

**Review by dnmeid — APPROVED** [review]

**Comment by Raphkitue** [comment]
All comments have been applied, feel free to merge it when it is convenient

**Pull request merged**